spec/unit/backend/hocon_backend_spec.rb

```diff
@@ -0,0 +1,134 @@
+            expect { @backend.lookup("key", {}, nil, :priority, nil) }.to raise_error(TypeError)
+          end
+      end
+
+      it "should build an array of all data sources for array searches" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({"key"=>"answer"})
+        @cache.expects(:read_file).with("/nonexisting/two.hocon", Hash).returns({"key"=>"answer"})
+        expect(@backend.lookup("key", {}, nil, :array, nil)).to eq(["answer", "answer"])
+      end
+
+      it "should ignore empty hash of data sources for hash searches" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({})
+        @cache.expects(:read_file).with("/nonexisting/two.hocon", Hash).returns({"key"=>{"a"=>"answer"}})
+        expect(@backend.lookup("key", {}, nil, :hash, nil)).to eq({"a" => "answer"})
+      end
+
+      it "should build a merged hash of data sources for hash searches" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({"key"=>{"a"=>"answer"}})
+        @cache.expects(:read_file).with("/nonexisting/two.hocon", Hash).returns({"key"=>{"b"=>"answer", "a"=>"wrong"}})
+        expect(@backend.lookup("key", {}, nil, :hash, nil)).to eq({"a" => "answer", "b" => "answer"})
+      end
+
+      it "should fail when trying to << a Hash" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({"key"=>["a", "answer"]})
+        @cache.expects(:read_file).with("/nonexisting/two.hocon", Hash).returns({"key"=>{"a"=>"answer"}})
+        expect {@backend.lookup("key", {}, nil, :array, nil)}.to raise_error(Exception, "Hiera type mismatch for key 'key': expected Array and got Hash")
+      end
+
+      it "should fail when trying to merge an Array" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({"key"=>{"a"=>"answer"}})
+        @cache.expects(:read_file).with("/nonexisting/two.hocon", Hash).returns({"key"=>["a", "wrong"]})
+
+        expect { @backend.lookup("key", {}, nil, :hash, nil) }.to raise_error(Exception, "Hiera type mismatch for key 'key': expected Hash and got Array")
+      end
```

Comment:
Since none of the tests above have any HOCON strings in them and don't end up actually parsing any HOCON data, in a way they are effectively duplicating the exact test coverage that is happening in the YAML one.  It might be worth DRY'ing the tests up and introducing some kind of 'common' backend test.  That might not really be worth the ROI, it seems like the JSON specs duplicate some of this too.  Something to think about but probably really necessary for now.